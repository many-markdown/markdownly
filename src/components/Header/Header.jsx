import React from 'react';
import {
  NavLink
} from 'react-router-dom';


export default function Header() {
  return (
    <>
      
      <NavLink to="/" >
        Landing Page
      </NavLink>
      <NavLink to="/AboutUS" >
        about Us
      </NavLink>
      <NavLink to="/Document" >
        Markdown
      </NavLink>

      
    </>
  );
}

