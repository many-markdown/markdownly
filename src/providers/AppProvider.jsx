import React, { useReducer, useEffect } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { setInitialState } from '../actions/markdownActions';

// eslint-disable-next-line react/prop-types
export default function AppProvider({ children }) {

  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    const initialState = localStorage.getItem('FILE_STORAGE'); 
    dispatch(setInitialState(JSON.parse(initialState)));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }} >
      {children}
    </AppContext.Provider>
  );
}
