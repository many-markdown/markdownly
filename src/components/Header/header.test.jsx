import React from 'react';
import { shallow } from 'enzyme';
import AppProvider from '../../providers/AppProvider';
import Header from './Header.jsx';

describe('SaveForm component', () => {
  it('renders SaveForm', () => {
    const wrapper = shallow(<AppProvider><Header /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
