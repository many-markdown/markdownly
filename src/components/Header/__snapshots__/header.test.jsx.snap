// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveForm component renders SaveForm 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "currentId": 1,
        "markdownFileList": Array [
          Object {
            "body": "# hello there",
            "fileName": "default",
            "id": 1,
          },
        ],
        "searchCategory": "fileName",
        "searchQuery": "",
      },
    }
  }
>
  <Header />
</ContextProvider>
`;
